File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ipconnection::IpConnection, linear_poti_bricklet::*};

const HOST: &str = "127.0.0.1";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Linear Poti Bricklet

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection
    let linear_poti_bricklet = LinearPotiBricklet::new(UID, &ipcon); // Create device object

    ipcon.connect(HOST, PORT).recv()??; // Connect to brickd
                                        // Don't use device before ipcon is connected

    // Get current position
    let position = linear_poti_bricklet.get_position().recv()?;
    println!("Position: {}", position); // Range: 0 to 100

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"tinkerforge/ipconnection"
	"tinkerforge/linear_poti_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Linear Poti Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	lp, _ := linear_poti_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current position.
	position, _ := lp.GetPosition()
	fmt.Printf("Position: \n", position) // Range: 0 to 100

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
